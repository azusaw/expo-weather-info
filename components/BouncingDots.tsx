import React, { useEffect, useRef } from "react";
import { Animated, View, StyleSheet } from "react-native";
import Colors from "@/constants/Colors";

const BouncingDots = () => {
  const dots = [
    useRef(new Animated.Value(0)).current,
    useRef(new Animated.Value(0)).current,
    useRef(new Animated.Value(0)).current,
  ];

  const colors = [
    Colors.secondary.light,
    Colors.secondary.default,
    Colors.secondary.dark,
  ];

  useEffect(() => {
    dots.forEach((dot, index) => {
      Animated.loop(
        Animated.sequence([
          Animated.delay(index * 150),
          Animated.timing(dot, {
            toValue: -10,
            duration: 300,
            useNativeDriver: true,
          }),
          Animated.timing(dot, {
            toValue: 0,
            duration: 300,
            useNativeDriver: true,
          }),
        ]),
      ).start();
    });
  }, []);

  return (
    <View style={styles.container}>
      {dots.map((dot, idx) => (
        <Animated.View
          key={idx}
          style={[
            styles.dot,
            { backgroundColor: colors[idx], transform: [{ translateY: dot }] },
          ]}
        />
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
  },
  dot: {
    width: 20,
    height: 20,
    marginHorizontal: 5,
    borderRadius: 5,
  },
});

export default BouncingDots;
