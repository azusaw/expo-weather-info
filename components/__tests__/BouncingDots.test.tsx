import { render } from "@testing-library/react-native";
import BouncingDots from "@/components/BouncingDots";

describe("BouncingDots", () => {
  test("it renders BouncingDots successfully", () => {
    const { toJSON } = render(<BouncingDots />);
    expect(toJSON()).toMatchSnapshot();
  });
});
