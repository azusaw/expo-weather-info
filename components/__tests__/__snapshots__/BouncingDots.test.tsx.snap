// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BouncingDots it renders BouncingDots successfully 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#fddab6",
        "borderRadius": 5,
        "height": 20,
        "marginHorizontal": 5,
        "transform": [
          {
            "translateY": 0,
          },
        ],
        "width": 20,
      }
    }
  />
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#ff9b5d",
        "borderRadius": 5,
        "height": 20,
        "marginHorizontal": 5,
        "transform": [
          {
            "translateY": 0,
          },
        ],
        "width": 20,
      }
    }
  />
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#ff6d1c",
        "borderRadius": 5,
        "height": 20,
        "marginHorizontal": 5,
        "transform": [
          {
            "translateY": 0,
          },
        ],
        "width": 20,
      }
    }
  />
</View>
`;
